src/olympia/blocklist/tests/test_cron.py

```diff
@@ -279,44 +526,71 @@ def test_upload_stash_unless_enough_changes(self):
+
+        self._block_version(is_signed=True, block_type=block_type)
+
         upload_mlbf_to_remote_settings()
         assert (
             mock.call(
                 self.current_time,
-                filter_list=[BlockType.BLOCKED.name],
+                filter_list=[block_type.name],
                 create_stash=False,
             )
             in self.mocks['olympia.blocklist.cron.upload_filter.delay'].call_args_list
         )
         new_mlbf = MLBF.load_from_storage(self.current_time)
-        assert new_mlbf.storage.exists(new_mlbf.filter_path())
+        assert new_mlbf.storage.exists(new_mlbf.filter_path(block_type))
         assert not new_mlbf.storage.exists(new_mlbf.stash_path)
 
-    def test_cleanup_old_files(self):
+    @mock.patch('olympia.blocklist.mlbf.BASE_REPLACE_THRESHOLD', 1)
+    def test_upload_stash_even_if_filter_is_updated(self):
         """
-        Cleanup old files only if a base filter already exists.
+        If enough changes of one type are made, update the filter, but still upload
+        a stash if there are changes of other types.
         """
-        upload_mlbf_to_remote_settings(force_base=True)
+        self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
+        self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
+        self._block_version(is_signed=True, block_type=BlockType.SOFT_BLOCKED)
+        upload_mlbf_to_remote_settings()
         assert self.mocks[
-            'olympia.blocklist.cron.cleanup_old_files.delay'
-        ].call_args_list == [mock.call(base_filter_id=self.base_time)]
+            'olympia.blocklist.cron.upload_filter.delay'
+        ].call_args_list == [
+            mock.call(
+                self.current_time,
+                filter_list=[BlockType.BLOCKED.name],
+                create_stash=False,
+            )
+        ]
+        mlbf = MLBF.load_from_storage(self.current_time)
+        assert mlbf.storage.exists(mlbf.filter_path(BlockType.BLOCKED))
+        assert not mlbf.storage.exists(mlbf.stash_path)
 
-        self.mocks[
-            'olympia.blocklist.cron.get_base_generation_time'
-        ].return_value = None
-        upload_mlbf_to_remote_settings(force_base=True)
-        assert (
-            self.mocks['olympia.blocklist.cron.cleanup_old_files.delay'].call_count == 1
-        )
+        with override_switch('enable-soft-blocking', active=True):
+            self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
+            self._block_version(is_signed=True, block_type=BlockType.BLOCKED)
+            upload_mlbf_to_remote_settings()
+            self.mocks['olympia.blocklist.cron.upload_filter.delay'].assert_called_with(
+                self.current_time,
+                filter_list=[BlockType.BLOCKED.name],
+                create_stash=False,
+            )
+            mlbf = MLBF.load_from_storage(self.current_time)
+            mlbf = MLBF.load_from_storage(self.current_time)
```

Comment:
Why do we generate twice?

Reply:
We generate zero times (akkkshully) but yeah we don't need to load from storage twice.